scripts/vet.sh

```diff
@@ -194,4 +194,16 @@ revive \
   -config "$(dirname "$0")/revive.toml" \
   ./...
 
+# Check dependencies of all non-internal and non-testing packages.
+OUTDIR="$(realpath "$(dirname "$0")")/deps"
+find . -name '*.go' | grep -v '/\(internal\|examples\|benchmark\|interop\|test\|testdata\)/' | xargs dirname | sort -u | while read d; do
+  pushd "$d"
+  pkg="$(echo "$d" | sed 's;^\./;;' | sed 's;\.;grpc;' | sed 's;/;_;g')"
+  go list -deps . | sort >| "${OUTDIR}/$pkg"
+  popd
+done
+
+git status --porcelain 2>&1 | fail_on_output || \
```

Comment:
We have two places where we currently call `git status --porcelain` and fail if there is non-empty output, and this is the third one. Can/should we make a helper?

Reply:
LGTM.